src/include/util/query_exec_util.h

```diff
@@ -222,6 +228,11 @@ class QueryExecUtil {
    */
   std::unordered_map<std::string, std::unique_ptr<planner::OutputSchema>> schemas_;
   std::unordered_map<std::string, std::unique_ptr<execution::compiler::ExecutableQuery>> exec_queries_;
+
+  /**
+   * Stores the most recently encountered error.
+   */
+  std::string error_msg_;
```

Comment:
Should this be reset in between new queries? So you must use it or lose it before invoking new queries. Seems like a source of potential confusion otherwise.

Reply:
Decided to reset the error when executing any operation that can throw (`PlanStatement`, `ExecuteDDL`, `CompileQuery`, and `ExecuteQuery`).